core/src/main/scala/cromwell/core/callcaching/docker/registryv2/flows/gcr/GoogleFlow.scala

```diff
@@ -0,0 +1,59 @@
+package cromwell.core.callcaching.docker.registryv2.flows.gcr
+
+import akka.stream.scaladsl.{Flow, GraphDSL, Merge, Partition}
+import akka.stream.{ActorMaterializer, FlowShape, ThrottleMode}
+import cromwell.core.callcaching.docker.DockerHashActor.{DockerHashContext, DockerHashResponse, DockerHashUnknownRegistry}
+import cromwell.core.callcaching.docker.registryv2.DockerRegistryV2AbstractFlow.HttpDockerFlow
+import cromwell.core.callcaching.docker.{DockerFlow, DockerImageIdentifierWithoutHash}
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+class GoogleFlow(httpClientFlow: HttpDockerFlow, queriesPer100Sec: Int)(implicit ec: ExecutionContext, materializer: ActorMaterializer) extends DockerFlow {
+  
+  private val gcrFlow = new GcrFlow(httpClientFlow)
+  private val usGcrFlow = new GcrUsFlow(httpClientFlow)
+  private val euGcrFlow = new GcrEuFlow(httpClientFlow)
+  
+  private val googleFlows = List(gcrFlow, usGcrFlow, euGcrFlow)
+  
+  // Throttle all requests to GCR
+  private val throttler = Flow[DockerHashContext].throttle(queriesPer100Sec, 100 seconds, queriesPer100Sec, ThrottleMode.Shaping)
+
+  override def buildFlow() = GraphDSL.create() { implicit builder =>
+    import GraphDSL.Implicits._
+
+    // This flow should only receive images that have been vetted by accepts.
+    def mapContextToFlow(dockerHashContext: DockerHashContext) = {
+      if (gcrFlow.accepts(dockerHashContext.dockerImageID)) 0
+      else if (usGcrFlow.accepts(dockerHashContext.dockerImageID)) 1
+      else if (usGcrFlow.accepts(dockerHashContext.dockerImageID)) 2
+      else 3
```

Comment:
any reason you couldn't use the "index of accepting object in the collection" magic like in the other spot?